src/Query.php

```diff
@@ -14,6 +14,58 @@ class Query
     protected $type;
     protected $wheres = [];
     protected $options = [];
+    protected $cacheQueryResults = false;
+    protected static $cachedRecords = [];
```

Comment:
Record or Document?

Reply:
I was thinking this, as at the moment they are really documents.